new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.assertions.impl;
+
+import java.util.ArrayList;
+import java.util.concurrent.Callable;
+
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  /**
+   * Check only one pod is restarted at a time in the same order as in the pods list.
+   * This assertion method needs to be called right after the domain is patched to ensure
+   * it doesn't miss any of the pods restart.
```

Comment:
On this statement: "This assertion method needs to be called right after the domain is patched to ensure it doesn't miss any of the pods restart."...  This seems likely to intermittently fail. I think it would be better to collect the current V1PodList in advance of starting the roll and using that as the input.

Reply:
+1 to that

Reply:
The test supplies the list of pods with its creation timestamps(created before the patching).